octoprint_mrbeam/iobeam/laserhead_handler.py

```diff
@@ -407,3 +409,65 @@ def _write_laser_heads_file(self, laser_heads_file=None):
                 self._logger.info("Writing to file: {} ..is successful!".format(laser_heads_file))
         except IOError as e:
             self._logger.error("Can't open file: {} , Due to error: {}: ".format(laser_heads_file, e))
+
+    @property
+    def current_laserhead_max_temperature(self):
+        """
+        Return the current laser head max temperature
+        Returns:
+        Laser head max temp (float)
+        """
+        current_laserhead_properties = self._load_current_laserhead_properties()
+
+        # Handle the exceptions
+        if((isinstance(current_laserhead_properties, dict) is False) or
+                ("max_temperature" not in current_laserhead_properties) or
+                (isinstance(current_laserhead_properties["max_temperature"], float) is False)):
+            # Apply fallback
+            self._logger.debug("Current laserhead properties: {}".format(current_laserhead_properties))
+            self._logger.exception(
+                "Current Laserhead Max temp couldn't be retrieved, fallback to the temperature value of: {}".format(
+                    LASERHEAD_MAX_TEMP_FALLBACK))
+            return LASERHEAD_MAX_TEMP_FALLBACK
+        # Reaching here means, everything looks good
+        self._logger.debug("Current Laserhead Max temp:{}".format( current_laserhead_properties["max_temperature"]))
+        return current_laserhead_properties["max_temperature"]
+
+    def _load_current_laserhead_properties(self):
+        """
+        Loads the current detected laser head related properties and return them
+        Returns:
+            Current laser head properties (dict), (None) otherwise
+
+        """
+        # 1. get the ID of the current laser head
+        laserhead_id = self.get_current_used_lh_model_id()
+
+        # 2. Load the corresponding yaml file and return it's content
+        lh_properties_file_path = os.path.join("/home/pi/oprint/lib/python2.7/site-packages/octoprint_mrbeam",
```

Comment:
I would not hardcode the path, you could use the path of the package itselfe `plugin._basefolder`

Reply:
Thank you, was looking for this one :). Done